content/chainguard/migration/migrating-node.md

```diff
@@ -0,0 +1,207 @@
+
+The easiest way to search for packages is via apk tools. For example:
+
+```shell
+docker run -it --rm cgr.dev/chainguard/wolfi-base
+f273a9aa3242:/# apk update
+fetch https://packages.wolfi.dev/os/aarch64/APKINDEX.tar.gz
+ [https://packages.wolfi.dev/os]
+OK: 53914 distinct packages available
+f273a9aa3242:/# apk search cairo
+cairo-1.18.0-r1
+cairo-dev-1.18.0-r1
+cairo-gobject-1.18.0-r1
+cairo-static-1.18.0-r1
+cairo-tools-1.18.0-r1
+harfbuzz-8.4.0-r1
+harfbuzz-dev-8.4.0-r1
+pango-1.52.2-r1
+pango-dev-1.52.2-r1
+py3-cairo-1.26.0-r0
+py3-cairo-dev-1.26.0-r0
+```
+
+These packages can then be easily added to your Dockerfile.
+For more searching tips, check the [Searching for
+Packages](/chainguard/migration-guides/migrating-to-chainguard-images/#searching-for-packages)
+section of our base migration guide.
+
+## Differences to Docker Official Image
+
+If you are migrating from the [Docker Official Image](https://hub.docker.com/_/node) there are a
+few differences that are important to be aware of.
+
+ - Our images run as the `node` user with UID 65532 by default. If you need elevated privileges
+   for a task, such as installing a dependency, you will need to change to the root user. For
+   example add a `USER root` statement into a Dockerfile. For security reasons you should make
+   sure that the production application runs with a lower privilege user such as `node`.
+ - `WORKDIR` is set to `/app` which is owned by the `node` user.
+ - The Docker Official images have a "smart" entrypoint that interprets the CMD setting. So `docker
+   run -it node` will launch the Node.js interpreter but `docker run -it node /bin/sh` will launch a
+   shell. The latter does not work with Chainguard Images. In the non `-dev` images, there is no
+   shell to launch, and in the `-dev` images you will need to change the entrypoint e.g. `docker run
+   --entrypoint /bin/sh -it cgr.dev/chainguard/node`.
+ - The image has a defined `NODE_PORT=3000` environment variable which can be used by applications
+ - Our Node.js images include [dumb-init](https://github.com/Yelp/dumb-init) which can be used to
+   wrap the Node process in order to handle signals properly and allow for graceful shutdown. You
+   can use dumb-init by setting an entrypoint such as: `ENTRYPOINT ["/usr/bin/dumb-init", "--"]`
+ - In general there are many less libraries and utilities in the Chainguard Image. You may find that
+   your application has an unexpected dependency which needs to be added into the Chainguard Image.
+
+
+## Migration Example
+
+A typical Dockerfile for a Node.js application using the Docker Official Image may look something like this:
+
+```Docker
+FROM node:latest
+
+ENV NODE_ENV production
+
+WORKDIR /usr/src/app
+
+COPY package.json .
+RUN npm install
+USER node
+COPY . .
+
+EXPOSE 3000
+
+CMD npm start
+```
+
+Directly porting to Chainguard Images with the least number of changes results in this Dockerfile:
+
+```Docker
+FROM cgr.dev/chainguard/node:latest-dev
```

Comment:
I guess a part of me wants a link to an pp and package.json file, even if they're extremely minimal, just something to get these to run without errors. Adding them here does seem like a lot of clutter, though. In mine I linked to a repo that has things set up to work with needed app and requirements . I don't know, maybe it's overkill.

Reply:
I think it's a fair point. I was considering it but was having problems finding a minimal example.

Reply:
I've added this (https://github.com/chainguard-dev/cg-images-node-migration)